recognition/include/pcl/recognition/impl/linemod/line_rgbd.hpp

```diff
@@ -150,72 +151,69 @@ LineRGBD<PointXYZT, PointRGBT>::loadTemplates (const std::string &file_name, con
-      const PointXYZRGBA & p = template_point_cloud.points[j];
+  return (true);
+}
 
-      if (!std::isfinite (p.x) || !std::isfinite (p.y) || !std::isfinite (p.z))
-        continue;
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointXYZT, typename PointRGBT> BoundingBoxXYZ
+pcl::LineRGBD<PointXYZT, PointRGBT>::computeBoundingBox (PointCloud<PointXYZRGBA> & template_point_cloud)
+{
+  BoundingBoxXYZ bb;
+  bb.x = bb.y = bb.z = std::numeric_limits<float>::max ();
+  bb.width = bb.height = bb.depth = 0.0f;
+
+  Eigen::Vector3f geometric_center = Eigen::Vector3f::Zero ();
+  Eigen::Vector3f min_pos, max_pos;
+  min_pos.fill (std::numeric_limits<float>::max ());
+  max_pos.fill (std::numeric_limits<float>::lowest ());
+  std::size_t counter = 0;
+  for (std::size_t j = 0; j < template_point_cloud.size (); ++j) 
+  {
+    const PointXYZRGBA & p = template_point_cloud.points[j];
 
-      min_x = std::min (min_x, p.x);
-      min_y = std::min (min_y, p.y);
-      min_z = std::min (min_z, p.z);
-      max_x = std::max (max_x, p.x);
-      max_y = std::max (max_y, p.y);
-      max_z = std::max (max_z, p.z);
+    if (!isXYZFinite(p))
+      continue;
 
-      center_x += p.x;
-      center_y += p.y;
-      center_z += p.z;
-
-      ++counter;
+    for (std::size_t row=0; row<3; ++row)
+    {
+      min_pos[row] = std::min (min_pos[row], p.getVector3fMap ()[row]);
+      max_pos[row] = std::max (max_pos[row], p.getVector3fMap ()[row]);
     }
 
-    center_x /= static_cast<float> (counter);
-    center_y /= static_cast<float> (counter);
-    center_z /= static_cast<float> (counter);
+    geometric_center += p.getVector3fMap ();
 
-    bb.width  = max_x - min_x;
-    bb.height = max_y - min_y;
-    bb.depth  = max_z - min_z;
+    ++counter;
+  }
+  geometric_center /= static_cast<float> (counter);
 
-    bb.x = (min_x + bb.width / 2.0f) - center_x - bb.width / 2.0f;
-    bb.y = (min_y + bb.height / 2.0f) - center_y - bb.height / 2.0f;
-    bb.z = (min_z + bb.depth / 2.0f) - center_z - bb.depth / 2.0f;
+  Eigen::Vector3f bb_dim = max_pos - min_pos;
+  bb.width  = bb_dim[0];
+  bb.height = bb_dim[1];
+  bb.depth  = bb_dim[2];
 
-    for (std::size_t j = 0; j < template_point_cloud.size (); ++j)
-    {
-      PointXYZRGBA p = template_point_cloud.points[j];
+  Eigen::Vector3f diff_pos = min_pos - geometric_center;
+  bb.x = diff_pos[0];
+  bb.y = diff_pos[1];
+  bb.z = diff_pos[2];
 
-      if (!std::isfinite (p.x) || !std::isfinite (p.y) || !std::isfinite (p.z))
-        continue;
+  for (std::size_t j = 0; j < template_point_cloud.size (); ++j) 
+  {
+    PointXYZRGBA p = template_point_cloud.points[j];
 
-      p.x -= center_x;
-      p.y -= center_y;
-      p.z -= center_z;
+    if (!isXYZFinite(p))
+      continue;
 
-      template_point_cloud.points[j] = p;
-    }
+    p.getVector3fMap () -= geometric_center;
+
+    template_point_cloud.points[j] = p;
   }
```

Comment:
This isn't indicated by function name. I don't know how to convey the normalization of template cloud in the name